### PR TITLE
Add URLs for dashboard tabs

### LINKED ISSUE #3134: Use URLs for dashboard tabs
We could use URLs for the tabs of the dashboard, e.g. `/dashboard/{gallery,datasets,tasks,annotations}`.

### PR DESCRIPTION
**Pull request by MichaelBuessemeyer**
This PR adds paths to directly access a tab of the dashboard.

### URL of deployed dev instance (used for testing):
- [https://addurlfordashboard.webknossos.xyz](https://addurlfordashboard.webknossos.xyz)

### Steps to test:
- `.../dashboard/gallery` should open the `dataset gallery`
- `.../dashboard/datasets` should open the `dataset-list view`
- `.../dashboard/tasks` should open the `tasks view`
- `.../dashboard/annotations` should open the `explorative annotations view`

### Issues:
- fixes #3134

------
- [x] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)
- ~~~[ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~~
- ~~~[ ] Updated [documentation](../blob/master/docs) if applicable~~~
- ~~~[ ] Needs datastore update after deployment~~~
- [x] Ready for review

### CODE REVIEW

File: app/assets/javascripts/router.js [hunk 1]
```diff
@@ -121,16 +121,49 @@ class ReactRouter extends React.Component<Props> {
...
                     <SpotlightView />
                   )
                 }
               />
+              <SecuredRoute
+                isAuthenticated={isAuthenticated}
+                path="/dashboard/:tab"
+                render={({ match }: ContextRouter) => {
+                  let initialTabKey: ?string;
+                  switch (match.params.tab) {
```
Comment (philippotto) [inline app/assets/javascripts/router.js]: The switch case here implements a mapping from url token to tab key which can be expressed more concisely using a simple object. Something like this should work:

```
const urlTokenToTabKeyMap = {
  gallery: "datasets",
  datasets: "advanced-datasets",
  tasks: "tasks",
  annotations: "explorativeAnnotations"
}
const initialTabKey = urlTokenToTabKeyMap[match.params.tab];
```

File: app/assets/javascripts/dashboard/dashboard_view.js [hunk 2]
```diff
@@ -21,11 +21,13 @@ import { parseAsMaybe } from "libs/utils";
 const TabPane = Tabs.TabPane;
 
 const validTabKeys = ["datasets", "advanced-datasets", "tasks", "explorativeAnnotations"];
+const tabKeyToURL = ["gallery", "datasets", "tasks", "annotations"];
```
Comment (philippotto) [inline app/assets/javascripts/dashboard/dashboard_view.js]: See my other comment for how to define the mapping with an object. If you need both directions (which seems to be the case, see line 192), you can use `_.invert` (https://lodash.com/docs/4.17.10#invert) to construct the "inverse mapping" object.

File: app/assets/javascripts/router.js (hunk 1 above)
Comment (philippotto) [reply app/assets/javascripts/router.js]: The mapping object should probably be defined in the dashboard file (where the other variables are defined). If you need it in this file, you can just import it.

**Comment by MichaelBuessemeyer** [comment]
@philippotto This PR is ready for another review

File: app/assets/javascripts/router.js [hunk 3]
```diff
@@ -121,16 +121,34 @@ class ReactRouter extends React.Component<Props> {
...
                     <SpotlightView />
                   )
                 }
               />
+              <SecuredRoute
+                isAuthenticated={isAuthenticated}
+                path="/dashboard/:tab"
+                render={({ match }: ContextRouter) => {
+                  const tab: ?string = match.params.tab;
+                  const initialTabKey: ?string = tab ? urlTokenToTabKeyMap[tab] : null;
```
Comment (philippotto) [inline app/assets/javascripts/router.js]: Are the two type annotations necessary for flow to succeed?

File: app/assets/javascripts/router.js [hunk 4]
```diff
@@ -121,16 +121,34 @@ class ReactRouter extends React.Component<Props> {
...
+                  return (
+                    <DashboardView
+                      userId={null}
+                      isAdminView={false}
+                      initialTabKey={initialTabKey || null}
```
Comment (philippotto) [inline app/assets/javascripts/router.js]: I think the `|| null` part can be removed?

Comment (MichaelBuessemeyer) [reply app/assets/javascripts/router.js]: Correct, didn't know that `undefined || "default"` results in `"default"` :slightly_smiling_face:

**Pull request merged**